R/class-KPartiteNetwork.R

```diff
@@ -116,4 +116,75 @@ KPartiteNetwork <- setClass("KPartiteNetwork",
     partitions = Partitions()
   ),
   validity = check_kpartite_network
-)
\ No newline at end of file
+)
+
+#' @include utils.R
+#' Generate a K-Partite Network 
+#' 
+#' Generate a K-Partite Network from a LinkList and NodeList, or from a data.frame
+#' @param links LinkList
+#' @param nodes NodeList
+#' @param object Object containing data to be converted to a Network
+#' @return KPartiteNetwork
+#' @export
+#' @examples
+#' KPartiteNetwork(data.frame(source='a',target='b'))
+setGeneric("KPartiteNetwork", 
+  function(
+    links,
+    nodes,
+    object, 
+    partitions = Partitions(),
+    linkColorScheme = 'none', 
+    variables = VariableMetadataList(), 
+    ...
+  ) standardGeneric("KPartiteNetwork"),
+  signature = c("links", "nodes", "object")
+)
+
+#' @export
+setMethod("KPartiteNetwork", signature("LinkList", "NodeList", "missing"), function(
+  links, 
+  nodes,
+  object,
+  partitions = Partitions(),
+  linkColorScheme = 'none', 
+  variables = VariableMetadataList(), 
+  ...
+) {
+  # default to a single partition if none provided
+  if (length(partitions) == 0) {
+    partitions <- Partitions(list(NodeIdList(nodes)))
+  }
+  new("KPartiteNetwork", links=links, nodes=nodes, partitions=partitions, linkColorScheme=linkColorScheme, variableMapping=variables)
+})
+
+#' @export  
+setMethod("KPartiteNetwork", signature("missing", "missing", "data.frame"), function(
+  links, 
+  nodes,
+  object = data.frame(source=character(),target=character()), 
+  partitions = Partitions(),
+  linkColorScheme = 'none', 
+  variables = VariableMetadataList(), 
+  ...
+) {
+  # default to a single partition if none provided
+  if (length(partitions) == 0) {
+    partitions <- Partitions(list(NodeIdList(object)))
+  }
+  new("KPartiteNetwork", links=LinkList(object), nodes=NodeList(object), partitions=partitions, linkColorScheme=linkColorScheme, variableMapping=variables)
```

Comment:
Would be helpful to also have a method taking a Network and Partitions to create a KPartiteNetwork.